1

In [1]:
def square_element(x):
    return x * x
test_list = [i for i in range(10)]
list(map(square_element, test_list))

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]

In [2]:
def square_list(list):
    new = []
    for i in list:
        new.append(i**2)
    return new
square_list(range(10))

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]

In [3]:
def listoc(listic): 
    new_listoc = []
    return list(map(lambda x: x**2, listic))
listoc(range(10))

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]

2

In [4]:
%%timeit
#для 10
list(map(square_element, test_list))

1.04 µs ± 47.9 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)


In [5]:
def square_element(x):
    return x * x
test_list = [i for i in range(100)]
list(map(square_element, test_list))

[0,
 1,
 4,
 9,
 16,
 25,
 36,
 49,
 64,
 81,
 100,
 121,
 144,
 169,
 196,
 225,
 256,
 289,
 324,
 361,
 400,
 441,
 484,
 529,
 576,
 625,
 676,
 729,
 784,
 841,
 900,
 961,
 1024,
 1089,
 1156,
 1225,
 1296,
 1369,
 1444,
 1521,
 1600,
 1681,
 1764,
 1849,
 1936,
 2025,
 2116,
 2209,
 2304,
 2401,
 2500,
 2601,
 2704,
 2809,
 2916,
 3025,
 3136,
 3249,
 3364,
 3481,
 3600,
 3721,
 3844,
 3969,
 4096,
 4225,
 4356,
 4489,
 4624,
 4761,
 4900,
 5041,
 5184,
 5329,
 5476,
 5625,
 5776,
 5929,
 6084,
 6241,
 6400,
 6561,
 6724,
 6889,
 7056,
 7225,
 7396,
 7569,
 7744,
 7921,
 8100,
 8281,
 8464,
 8649,
 8836,
 9025,
 9216,
 9409,
 9604,
 9801]

In [6]:
%%timeit
#для 100
list(map(square_element, test_list))

9.63 µs ± 1.31 µs per loop (mean ± std. dev. of 7 runs, 100000 loops each)


In [7]:
%%timeit
listoc(range(10))

3.81 µs ± 145 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)


In [11]:
%%timeit
listoc(range(100))

36.3 µs ± 1.35 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)


Выводы : скорость зависит от количества подаваемых на вход чисел, но время прогона в функции с использованием цикла (for) - для 100 чисел составило 9.63 µs ± 1.31 , тогда как для того же числа входных данных - 36.3 µs ± 1.35 µs per loop 

3

In [9]:
def mean(a):
    sum = 0
    for elem in num_list:
        sum += elem
    return sum / len(a)
def median(a):
    a.sort()
    x = int(len(a) // 2)
    x1 = int(len(a) / 2 - 1)
    x2 = int(len(a) / 2)
    if len(a) % 2 != 0:
        return num_list[x]
    return ((num_list[x1] + num_list[x2]) / 2)

In [10]:
import numpy as np
keys = ['mean', 'median', 'mode']
a = [int(x) for x in input().split()]
mylist=[]
mylist.append(np.mean(a))
mylist.append(np.median(a))
mylist.append(max(set(a), key = a.count))
dictionary = dict(zip(keys, mylist))
print(dictionary) 

 3 3 4 3


{'mean': 3.25, 'median': 3.0, 'mode': 3}
